lib/src/components/AbstractDialog/AbstractDialog.tsx

```diff
@@ -0,0 +1,213 @@
+     *
+     * Note: not supported in safari, which is why the conditional is here. Once safari gets
+     * it together, we should be able to take advantage of the "top layer" functionality built into
+     * the browser around modals to simplify the dialog styling around z-index handling
+     */
+    if (hasNativeModalSupport) {
+      // https://github.com/microsoft/TypeScript-DOM-lib-generator/issues/1029#issuecomment-968299542
+      // eslint-disable-next-line @typescript-eslint/no-explicit-any
+      (el as any).showModal();
+    }
+    else {
+      // without native support we don't trap focus in the dialog, but we can at least start it off there
+      el.focus();
+    }
+
+    return () => {
+      if (previouslyFocusedEl && previouslyFocusedEl instanceof HTMLElement) {
+        // The useEffect cleanup executes while the dialog is still present (in React 16 at least). While the dialog
+        // still exists, the document is still "blocked by the modal dialog" so trying to focus elements outside of
+        // it won't work. So we have to wait until the next cycle of the event loop when it's gone
+        Promise.resolve().then(() => previouslyFocusedEl.focus());
+      }
+    };
+  }, []);
+
+  // listen to the native HTMLDialogElement cancel event which supporting browsers fire when the dialog is closed
+  // via ESC
+  useEffect(function() {
+    const dialog = dialogRef.current;
+
+    if (dialog && hasNativeModalSupport) {
+      /* eslint-disable @typescript-eslint/no-non-null-assertion */
+      dialog.addEventListener('cancel', onCancel!);
+      return () => { dialog.removeEventListener('cancel', onCancel!); };
+      /* eslint-enable @typescript-eslint/no-non-null-assertion */
+    }
+    else {
+      return undefined;
+    }
+  }, [onCancel]);
+
+  if (cancelOnClickOutside) {
+    const clickOutsideTargetElement = cancelOnClickOutsideTargetClassName ?
+      dialogRef.current?.getElementsByClassName(cancelOnClickOutsideTargetClassName)[0] :
+      dialogRef.current;
+
+    useEffect(() => {
+      const listener = (event: MouseEvent) => {
+        if (!clickOutsideTargetElement || clickOutsideTargetElement.contains(event.target as Node)) {
+          return;
+        }
+
+        if (onCancel) {
+          onCancel(createCancelEvent());
+        }
+      };
+
+      document.addEventListener('click', listener);
+
+      return () => document.removeEventListener('click', listener);
+    }, [clickOutsideTargetElement]);
+  }
+
+  const attributesWithoutTabIndex = omit(['tabIndex'], attrs);
+
+  const abstractDialogContextValue = {
+    dialogEl: dialogRefState,
+    onCancel: onCancel
+  };
+
+  return (
+    // Provide the dialog element to descendants so that tooltips can attach to it instead of the body,
+    // which is necessary so that they end up in the top layer rather than behind the modal
+    <AbstractDialogContext.Provider value={abstractDialogContextValue}>
```

Comment:
I'm not sure it's useful to the context consumers for "Abstract" to be in its name.  How about just `DialogContext`?

Reply:
Renamed here:
e7b2fe0f3c4fac09770f40c9493a619a24f4fa19